app/code/Magento/QuoteGraphQl/Model/Resolver/DataProvider/Cart/CartHydrator.php

```diff
@@ -0,0 +1,47 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\QuoteGraphQl\Model\Resolver\DataProvider\Cart;
+
+use Magento\Quote\Api\Data\CartInterface;
+use Magento\Quote\Model\Quote;
+use Magento\Quote\Model\Quote\Item as QuoteItem;
+
+/**
+ * Cart Hydrator class
+ */
+class CartHydrator
+{
+    /**
+     * Hydrate cart to plain array
+     * @param CartInterface|Quote $cart
+     * @return array
+     */
+    public function hydrate(CartInterface $cart): array
```

Comment:
In terms of classic mean of hydrator I would name this method as `extract`. The `hydrate` makes the opposite action: casts array to an object